ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh

```diff
@@ -78,6 +79,17 @@ else
   fi
   subfix=$(openssl rand -hex 2)
   CLUSTER_NAME=${CLUSTER_NAME:-"$prefix-$subfix"}
+  # For long cluster name enabled, append a "long_name_prefix_len" chars long random string to cluster name
+  # Max possible prefix length is 9, max possible subfix length is 4, hyppen is used 2 times, random string of length "long_name_prefix_len"
+  # (9 + 4 + 2 + "long_name_prefix_len" = 54 )
+  if [[ "$LONG_CLUSTER_NAME_ENABLED" == "true" ]]; then
+    long_name_prefix_len=$(( 54 - $(echo -n "$CLUSTER_NAME" | wc -c) - 1 ))
+    long_name_prefix=$(head /dev/urandom | tr -dc 'a-z0-9' | head -c $long_name_prefix_len)
+    CLUSTER_NAME="$prefix-$subfix-$long_name_prefix"
+    #set the domain prefix of length (<=15)
+    DOMAIN_PREFIX="$prefix-$subfix"
+    DOMAIN_PREFIX_SWITCH="--domain-prefix $DOMAIN_PREFIX"
```

Comment:
In the case of using a long name, the domain prefix does not have to be manually set. If we don't set one, system will automatically generate a random 15chars domain prefix. 
With current test code, we'll not cover the domain prefix auto-generation code path.

How about add a similar var as 'LONG_CLUSTER_NAME_ENABLED', let's say 'SPECIFY_DOMAIN_PREFIX' default to false. 
and only when it is 'true', we do something like:
```
if [[ "$SPECIFY_DOMAIN_PREFIX" == "true" ]]; then
    DOMAIN_PREFIX="$prefix-$(tr -dc 'a-z0-9' < /dev/urandom | head -c $MAX_DOMAIN_PREFIX_LENGTH)"
...
```
And only set it to 'true' in some jobs

Reply:
fixed